### PR TITLE
move xslt transforms into ingester app

### PR DESCRIPTION
**Pull request by andygnewman**
To Do

- [x] Move xslt transforms from article app
- [x] Make sure unit tests for each transform still work
- [x] Add promise into article v3 controller based on bodyXML and openingXML
- [x] Add to overall article model as bodyHTML and openingHTML
- [x] Write integration tests for bodyHTML and openingHTML

### CODE REVIEW

File: server/models/article-v3.js [hunk 1]
```diff
@@ -266,6 +267,13 @@ function cleanAndSortStrings (items) {
 		});
 }
 
+function contentHtmlTransform (contentXml, params) {
+	if (contentXml) {
+		return contentXsltTransform(contentXml, 'main', params);
+	}
+	return Promise.resolve();
```
Comment (i-like-robots) [inline server/models/article-v3.js]: Just don't call this method?

File: server/models/article-v3.js [hunk 2]
```diff
@@ -326,10 +334,27 @@ function applyModel (articleV1, articleV2) {
...
+	let bodyHTML = contentHtmlTransform(article.bodyXML, xsltParams).then(transformedHtml => {
+		if (transformedHtml) {
+			article.bodyHTML = transformedHtml;
+		}
+	});
+
+	let openingHTML = contentHtmlTransform(article.openingXML, xsltParams).then(transformedHtml => {
+		if (transformedHtml) {
+			article.openingHTML = transformedHtml;
+		}
```
Comment (i-like-robots) [inline server/models/article-v3.js]: I think the condition is a bit pointless, `article.openingHTML = undefined` won't do any harm.

```js
JSON.stringify({ param: undefined }) // => {}
```

File: server/models/article-v3.js [hunk 3]
```diff
@@ -326,10 +334,27 @@ function applyModel (articleV1, articleV2) {
 		}
 	});
 
+	let xsltParams = {
+		id: article.id,
+		webUrl: article.webUrl
+	};
+
+	let bodyHTML = contentHtmlTransform(article.bodyXML, xsltParams).then(transformedHtml => {
```
Comment (i-like-robots) [inline server/models/article-v3.js]: `const bodyHTML = article.bodyXML && contentHtmlTransform(...)` will do (see above)

File: tests/server/stylesheets/transform-helper.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+var contentXSLT = require('../../../server/stylesheets/content-xslt');
+
+module.exports = function(xml) {
+
+	return contentXSLT(xml, 'main');
+};
```
Comment (i-like-robots) [inline tests/server/stylesheets/transform-helper.js]: This helper isn't much help now?

**Comment by i-like-robots** [comment]
Awesome

**Comment by andygnewman** [comment]
Before we merge, do we need to do something to the build process for the libxslt? as with article?

**Comment by i-like-robots** [comment]
The XSLT tests are all green so it would appear not

**Comment by andygnewman** [comment]
That's a good point.

**Comment by matthew-andrews** [comment]
does this not require an update to the V3 elastic search index model?

**Comment by matthew-andrews** [comment]
^^ to answer my question, the answer is no :-)
https://github.com/Financial-Times/next-es-interface/blob/master/schema/content.json#L37-L48

**Comment by andygnewman** [comment]
Thanks to @i-like-robots for adding the new fields in during the reindexing.

**Comment by andygnewman** [comment]
@matthew-andrews can you help me understand what went wrong with this merge?

**Comment by matthew-andrews** [comment]
i'll pop over in a bit

**Pull request merged**